### DISCUSSION TITLE
Nuxt layer does not provide assets

### DISCUSSION THREAD

**Discussion by hellishcz** [nuxt/nuxt#19282] [Questions] [answered]
Hi, I am building internal component library for Nuxt 3 that I need to reuse in multiple project.

I created the library using Nuxt layer (https://nuxt.com/docs/getting-started/layers), but part of the library are assets (icons and global css styles). Everything works great, except that the assets are not available when I add the component library layer to another project using extends.

Is that a bug? Or if it is intentional, how should I create such assets in the component library? Or is there a better approach to component library than using layers?

Thanks for any advice!

**Comment by TheAlexLichter** [comment, marked as answer]
Hey! Can you provide a reproduction? **Reproduction was provided below**

From the [docs](https://nuxt.com/docs/guide/going-further/layers#relative-paths-and-aliases):
> When importing using aliases (such as ~/ and @/) in a layer components and composables, note that aliases are resolved relative to the user's project paths. As a workaround, you can use relative paths to import them. We are working on a better solution for named layer aliases.

## Solution

Use relative paths as described, so `<img src="~/assets/kitten.jpeg" />` should be `<img src="../assets/kitten.jpeg" />`

**Reply by hellishcz** [reply]
Great, thank you! I missed that part in the documentation.

**Reply by TheAlexLichter** [reply]
You are welcome 🙏

**Comment by hellishcz** [comment]
Would this be sufficient?

- Create Nuxt project and name it "component-library"
- Create one image in assets named image.jpg
- Create one component Component.vue in components directory. The component should show some text and then display the image.jpg from assets directory
- Create one page that uses the Component.vue.
- When launched and page opened, the Component should show the image.jpg

Now:
- Create another project
- Add extends: ['../component-library'] to the nuxt.config.ts
- Create one page in pages and use the Component on the page
- When launched and the page is shown, the Component shows it's text content, but the image is not found.

**Reply by TheAlexLichter** [reply]
It'd be ideal to put it in a StackBlitz or Codesandbox (see e.g. https://nuxt.new/)

**Reply by hellishcz** [reply]
Yes, that would be ideal. I tried that, but I failed, because I need to have both, the Layer and another Nuxt project that extends the Layer, in one StackBlitz instance. I was not able to create that :(

**Reply by TheAlexLichter** [reply]
You can create two top-level folders in stackblitz and do as you like 😋 
Or you use the layer starter with a playground folder https://stackblitz.com/github/nuxt/starter/tree/layer?file=README.md

**Reply by hellishcz** [reply]
Thanks for your help, I am new to StackBlitz!

Ok, here is my example. There is a layer with component that I can use in the app using extends. However, the component is using image from assets dir, and that is not found when embeded to the app.

https://stackblitz.com/edit/github-vlegsv?file=app.vue

**Reply by TheAlexLichter** [reply]
Updating my answer above so it can be marked 👍
